### ISSUE TITLE
Assist for adding From impls for enum variants

### ISSUE THREAD

**Issue by mattyhall** [rust-analyzer/rust-analyzer#3766]
For people who write their own error types, the following is quite a common pattern:

```rust
enum MyError {
    Read(std::io::Error),
    InvalidUtf8(std::str::Utf8Error),
}

impl std::convert::From<std::io::Error> for MyError { /* ... */ }
impl std::convert::From<std::std::Utf8Error> for MyError { /* ... */ }
```
It might be nice to have an assist for this as filling in the From impl is trivial in the above cases.

**Comment by matklad** [comment]
+1

The assist should trigger if the cursor is on a new-type style enum *variant*. Ie, to get the above picture, the assist should be invoked twice. 

A similar assist to look at is https://github.com/rust-analyzer/rust-analyzer/blob/1c2d4135db867efe335a0654d86429bea7bb9caf/crates/ra_assists/src/handlers/add_impl.rs

**Comment by mattyhall** [comment]
Should the implementation of this follow the same pattern of writing the impl into a string and inserting it as the code linked? I've seen there's an ast::make module but it seems quite limitted and is basically doing the same thing

**Comment by matklad** [comment]
For the time being I think either approach will work. 

Long term, we should move everything to `make`, and, for example, consolidate "add impl", "replace derive with a manual impl" and this assist.
